src/index.js

```diff
@@ -8,6 +8,20 @@ const supportMultiple = (typeof document !== 'undefined' && document && document
   true;
 
 class Dropzone extends React.Component {
+  static renderChildren(children, isDragActive, isDragReject) {
+    if (!children) {
+      return children;
+    }
+    let renderedChildren = children;
+    const props = { isDragActive, isDragReject };
+    if (typeof children === 'function') {
+      renderedChildren = children(props);
```

Comment:
so if you provide a function, the props will not be added to the children. In case some people start mixing together the two syntax, it might seem weird to them ... But you can decide that you should chose between the two syntaxes.